pkg/telemetry/phonehome/segment/segment.go

```diff
@@ -89,14 +88,39 @@ func (t *segmentTelemeter) Stop() {
 	}
 }
 
-func (t *segmentTelemeter) Identify(userID string, props map[string]any) {
+func (t *segmentTelemeter) overwriteID(id string) string {
+	if id == "" {
+		return t.userID
+	}
+	return id
+}
+
+func makeDeviceContext(clientID, clientType string) *segment.Context {
+	if clientID == "" {
+		return nil
+	}
+	return &segment.Context{
+		Device: segment.DeviceInfo{
+			Id:   clientID,
+			Type: clientType,
+		},
+	}
+}
+
+func (t *segmentTelemeter) Identify(props map[string]any) {
+	t.IdentifyUserAs("", "", "", props)
+}
+
+func (t *segmentTelemeter) IdentifyUserAs(userID, clientID, clientType string, props map[string]any) {
```

Comment:
It's a bit unfortunate that we have the parameters but their usage varies based on the context of callers (i.e. sometimes callers will explicitly set `clientID, clientType`, other times they will leave it empty).

I'm also not quite sure, just reading through the code, when its OK to leave those things empty and when not, I suppose.

For the clarity on which parameters to use, albeit it might become overkill, but will allow us to extend the API further in the future, how about creating `Options` that can be passed to the `*UserAs` methods, so its more explicit what parameters are used and we get rid of the "empty string" parameters (since go doesn't allow for default parameter values)?
As an example, something similar to[ the auth provider options we have](https://github.com/stackrox/stackrox/blob/1b934bf19e1f60a23f3e0f35dbcde15584e29c24/pkg/auth/authproviders/provider_options.go#L20) would make it more explicit which values to use.

Reply:
Any change like this would imply a change in the interface, and Segment implementation doesn't import it, and cannot implement. That's unfortunate.
A possible solution would be to extract the interface to a third package. I'll see if that will look good and go-ish enough.